### PR TITLE
Calling ocrValidationRetryManager.isReadyToRetry from BlobProcessorTask

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-709

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ProcessorTestSuite.java [hunk 1]
```diff
@@ -80,6 +81,9 @@
     @Autowired
     protected ProcessEventRepository processEventRepository;
 
+    @Autowired
+    OcrValidationRetryManager ocrValidationRetryManager;
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ProcessorTestSuite.java:85]: access modifier

Comment (apereverzin) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ProcessorTestSuite.java]: `protected` are actually not necessary, maybe worth removing all of them in one separate PR.

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java [hunk 2]
```diff
@@ -157,11 +158,14 @@ public void setup() {
             leaseMetaDataChecker
         );
 
+        OcrValidationRetryManager ocrValidationRetryManager = new OcrValidationRetryManager(2, 300);
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java:161]: shouldnt this be mocked/autowired like all other dependencies?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java [hunk 3]
```diff
@@ -122,11 +127,14 @@ private void leaseAndProcessZipFile(
 
         leaseAcquirer.ifAcquiredOrElse(
             blobClient,
-            leaseId -> processZipFile(container, blobClient, zipFilename, leaseId),
+            leaseId -> {
+                if (ocrValidationRetryManager.isReadyToRetry(blobClient)) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java:131]: Method name seems a bit weird, are we retrying here or is it the first attempt?
If it doesn't matter then perhaps this should be renamed?

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java [hunk 4]
```diff
@@ -58,21 +64,23 @@ void setUp() {
...
             fileContentProcessor,
-            leaseAcquirer
+            leaseAcquirer,
+            ocrValidationRetryManager
         );
     }
 
     @Test
-    void processBlobs_should_not_call_envelopeProcessor_if_failed_to_load_file() throws Exception {
+    void processBlobs_should_not_call_envelopeProcessor_if_failed_to_load_file() {
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java:73]: unrelated

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java [hunk 5]
```diff
@@ -58,21 +64,23 @@ void setUp() {
...
         // given
         given(blobManager.listInputContainerClients()).willReturn(singletonList(container));
 
         PagedIterable<BlobItem> pagedIterable = mock(PagedIterable.class);
         given(container.listBlobs()).willReturn(pagedIterable);
-        given(pagedIterable.stream()).willReturn(singletonList(blob).stream());
+        given(pagedIterable.stream()).willReturn(Stream.of(blob));
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java:79]: unrelated

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java (hunk 3 above)
Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java]: Renamed to `canProcess`

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java (hunk 2 above)
Comment (apereverzin) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeControllerTest.java]: autowired

**Review by aliveni — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java [hunk 6]
```diff
@@ -122,11 +127,14 @@ private void leaseAndProcessZipFile(
 
         leaseAcquirer.ifAcquiredOrElse(
             blobClient,
-            leaseId -> processZipFile(container, blobClient, zipFilename, leaseId),
+            leaseId -> {
+                if (ocrValidationRetryManager.canProcess(blobClient)) {
+                    processZipFile(container, blobClient, zipFilename, leaseId);
+                }
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java:133]: it may be worth logging something if it's not ready yet imho

**Review by rkondratowicz — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java (hunk 6 above)
Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java:133]: It is logged in `OcrValidationRetryManager`
